Parsers/ASimAuditEvent/Parsers/vimAuditEventSentinelOne.yaml

```diff
@@ -0,0 +1,714 @@
+            | parse-kv DataFields_s as (username: string, userName: string, userFullName: string, newValue: string, policyEnabled: string, siteName: string, oldValue: string, ipAddress: string, oldSiteName: string, policy: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
+            | parse-kv policy as (id: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
+            | project-rename ObjectId = id
+            | lookup EventFieldsLookup on activityType_d;
+        let groupsiteactivitydata_onoff = rawgroupsiteactivitydata
+            | where activityType_d in(39, 41, 57, 105, 200, 73, 76, 78, 79, 84, 87, 150)
+            | lookup EventTypeLookup_onoff on $left.newValue == $right.field
+            | lookup EventTypeLookup_onoff on $left.policyEnabled == $right.field
+            | extend
+                EventType = coalesce(EventType_field, EventType_field1),
+                NewValue = coalesce(NewValue_field, NewValue_field1);
+        let groupsiteactivitydata_enabledisabled = rawgroupsiteactivitydata
+            | where activityType_d in (70, 82, 83, 201)
+            | lookup EventTypeLookup_enableddisabled on $left.newValue == $right.field
+            | extend
+                EventType = EventType_fieldenableddisabled,
+                NewValue = NewValue_fieldenableddisabled;
+        let groupsiteactivitydata_other = rawgroupsiteactivitydata
+            | where activityType_d !in(39, 41, 57, 105, 200, 73, 76, 78, 79, 84, 87, 150, 70, 82, 83, 201)
+            | extend EventType = EventType_activity;
+        let groupsiteactivitydata = union
+                groupsiteactivitydata_onoff,
+                groupsiteactivitydata_enabledisabled,
+                groupsiteactivitydata_other
+            | extend
+                ActorUsername = coalesce(username, userName, userFullName),
+                Object = coalesce(Object, siteName, oldSiteName),
+                NewValue = coalesce(NewValue, newValue),
+                OldValue = oldValue;
+        let machineactivitydata = activitydata
+            | where activityType_d in (52, 53, 54, 55, 61, 62, 63, 93, 95, 117, 118, 4100, 4101)
+            | parse-kv DataFields_s as (username: string, userName: string, computerName: string, threatClassification: string, ipAddress: string, groupName: string, targetGroupName: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
+            | lookup EventFieldsLookupMachineActivity on activityType_d
+            | extend
+                EventType = EventType_machineactivity,
+                EventSubType = EventSubType_machineactivity,
+                ThreatCategory = threatClassification,
+                OldValue = groupName,
+                NewValue = targetGroupName,
+                ObjectId = agentId_s
```

Comment:
The id of the object on which the operation identified by EventType is performed? AgentId_s doesn't seem like object on which operation is perfomed? In this there should be groupid or something?

Reply:
got it